manager/integration/tests/test_recurring_job.py

```diff
@@ -2069,3 +2075,96 @@ def test_recurring_job_restored_from_backup_target(set_random_backupstore, clien
     wait_for_volume_recurring_job_update(rvolume2,
                                          jobs=[snap1, back1],
                                          groups=[DEFAULT, group1])
+
+
+@pytest.mark.recurring_job  # NOQA
+def test_recurring_job_filesystem_trim(client, core_api, batch_v1_api, volume_name, csi_pv, pvc, pod_make):  # NOQA
+    """
+    Scenario: test recurring job filesystem-trim
+
+    Given a workload (volume, pv, pvc, pod).
+    And create 50mb file in volume.
+    And actual size of the volume should increase.
+    And delete the 50mb file in volume.
+    And actual size of the volume should not decrease.
+
+    When create a recurring job with:
+         - task: filesystem-trim
+         - retain: 1
+    Then recurring job retain mutated to 0.
+
+    When assign the recurring job to volume.
+    And wait for the cron job scheduled time.
+    Then volume actual size should decrease 50mb.
```

Comment:
This is a good example of BDD. Of course, some parts need to adjust for a better keyword statement. 

cc @longhorn/qa